jax/numpy/lax_numpy.py

```diff
@@ -3787,3 +3788,115 @@ def _unstack(x):
     raise ValueError("Argument to _unstack must be non-scalar")
   return [lax.index_in_dim(x, i, keepdims=False) for i in range(x.shape[0])]
 setattr(DeviceArray, "_unstack", _unstack)
+
+
+# Syntactic sugar for scatter operations.
+class IndexUpdateHelper(object):
+  # Note: this docstring will appear as the docstring for the `at` property.
+  """Indexable helper object to call indexed update functions.
+
+  The `at` property is syntactic sugar for calling the indexed update functions
+  defined in :mod:`jax.ops`, and acts as a pure equivalent of in-place
+  modificatons.
+
+  In particular:
+  - :code:`x = x.at[idx].set(y)` is a pure equivalent of :code:`x[idx] = y`.
```

Comment:
Another valid option is to use double ticks here, e.g.,
```
``x = x.at[idx].set(y)``
```

Or actually I guess this doesn't really need to be valid RST at all if we aren't putting it in the docs somewhere.

Reply:
True. I guess the only time someone would see this would be if they were either reading the source code or using  `help` in a REPL.

I've changed it to double ticks because that's easier to read. I suppose single ticks (or even no ticks) would also work, but I'm not sure it really matters; people reading this can probably figure it out either way.